pkg/storage/system_intake_funding_source.go

```diff
@@ -13,13 +13,10 @@ import (
 )
 
 func (s *Store) UpdateSystemIntakeFundingSources(ctx context.Context, systemIntakeID uuid.UUID, fundingSources []*models.SystemIntakeFundingSource) ([]*models.SystemIntakeFundingSource, error) {
```

Comment:
Also not in the scope of the changes for this PR, but I think WithTransaction blocks should ideally happen one level higher than these storage package calls. 

If you pass a NamedPreparer, this could be a atomic, or part of a larger transaction. As it stands, most of these methods have to be atomic

Reply:
Our thinking with this pattern is that it allows the caller to use this functionality even if not carrying a `tx` along with it

Reply:
Yeah! So, a named preparer accomplishes that, without limiting the transaction to the store method. Right now, we have two implementations of the named preparer interface, `store` and `tx`. 

So if you pass store as the named preparer you can still use the functionality without being in a transaction block. BUT it also allows the greater flexibility to include it as part of a larger transaction.

Reply:
Ahhh - I see!! Thanks for explaining that! I will re-look at that in a separate branch off of this one! Thanks Steven 😄

Reply:
@StevenWadeOddball I think Sam might have done this approach to avoid having to change every call to `UpdateSystemIntakeFundingSources` (and replace it with one that takes a namedpreparer, like `UpdateSystemIntakeFundingSourcesNP` (kinda) does)

@samoddball Perhaps it makes the most sense to see if it's just easier to replace calls from `UpdateSystemIntakeFundingSources` to `UpdateSystemIntakeFundingSourcesNP`, since it only looks like there's a small amount!

Reply:
Sounds good! Also to call out, I don't mean this needs to happen in the PR, but I wanted to provide this thought! 

I think ideally you'd only need one function that takes a named preparer (or expand the interface if you need to do named exec etc). But that's not in scope of the change for this PR, but I think it's worth keeping in mind as tech debt.

Reply:
Agree, not needed in this PR!

For context, @StevenWadeOddball, we also had opted to use a `sqlx.Tx` instead of a `NamedPreparer` in some spots because we call multiple SQL calls (like [this](https://github.com/CMSgov/easi-app/blob/996b380228a87238a0d4b033a252d45803aa09d4/pkg/storage/system_intake_funding_source.go#L66-L69)) and didn't want to offer the capacity for that to be done outside a transaction (Perhaps we should just rename that function at some point from `UpdateSystemIntakeFundingSourcesNP` to `UpdateSystemIntakeFundingSourcesTX`)

Reply:
I think I will look again at this in a follow-on. Naively just changing the type from `sqlx.Tx` -> `sqlutils.NamedPreparer` (and updating args to be `map[string]any`) results in some oddities - will look again

Reply:
Yeah, plus using a `NamedPreparer` here opens opportunities for using a `store` instead of `tx` here, which I'd advise against!

Reply:
That makes sens to me! Thank you both for the context.

Reply:
Ok thinking more about this, it seems `UpdateSystemIntakeFundingSources` should not be accessible from outside of the `storage` package

I think it should be unexported and not even a method on `Store`

```go
func updateSystemIntakeFundingSources(ctx context.Context, tx *sqlx.Tx, etc...)
```

and it should only ever be called from within the Update System Intake flow -- it doesn't seem like it should ever be called alone or even outside of a transaction

Reply:
^for discussion of course and, if agreed, another PR. Not this PR

Reply:
it can be treated as more of a convenience function rather than an actual store method